### PR TITLE
Add warning and make warnings louder

### PR DESCRIPTION
**Pull request by lukebakken**
Improve git describe command, add warning when on tag and `deps` target executed, make warnings more noticable in the output.

Example output - note that `riak_ts-0.0.2` was tagged incorrectly and should be removed:

```
(OTP_R16B02_basho10rc2)lbakken@brahms ~/Projects/basho/riak_develop (develop *=)
$ make devrel DEVNODES=5
Makefile:27: "Warning: you have checked out a tag (riak_ts-0.0.2) and should use the locked-deps target"
./rebar get-deps
```

Example when on an actual tag:

```
(OTP_R16B02_basho10rc2)lbakken@brahms ~/Projects/basho/riak_develop ((riak-2.1.3) *)
$ make devrel DEVNODES=5
Makefile:27: "Warning: you have checked out a tag (riak-2.1.3) and should use the locked-deps target"
./rebar get-deps
WARN:  Directory expected to be an app dir, but no app file found
in ebin/ or src/:
/home/lbakken/Projects/basho/riak_develop/deps/rebar_lock_deps_plugin
WARN:  Missing plugins: [rebar_lock_deps_plugin]
WARN:  Missing plugins: [cuttlefish_rebar_plugin,rebar_lock_deps_plugin]
==> rel (get-deps)
WARN:  Missing plugins: [rebar_lock_deps_plugin]
==> riak (get-deps)
==> riak_develop (get-deps)
WARN:  Directory expected to be an app dir, but no app file found
in ebin/ or src/:
/home/lbakken/Projects/basho/riak_develop/deps/rebar_lock_deps_plugin
Pulling node_package from {git,"git://github.com/basho/node_package.git",
                               {tag,"2.0.3"}}
Cloning into 'node_package'...
Pulling lager_syslog from {git,"git://github.com/basho/lager_syslog.git",
                               {branch,"2.1"}}
Cloning into 'lager_syslog'...
^CMakefile:27: recipe for target 'deps' failed
make: *** [deps] Interrupt
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,9 @@
 REPO            ?= riak
-PKG_REVISION    ?= $(shell git describe --tags)
+HEAD_REVISION   ?= $(shell git describe --tags --exact-match HEAD 2>/dev/null)
+PKG_REVISION    ?= $(shell git describe --tags 2>/dev/null)
 PKG_BUILD        = 1
 BASE_DIR         = $(shell pwd)
```
Comment (lukebakken) [inline Makefile:5]: This could be `$(realpath $(CURDIR))` to avoid a shell.

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
-ERLANG_BIN       = $(shell dirname $(shell which erl))
+ERLANG_BIN       = $(shell dirname $(shell which erl 2>/dev/null) 2>/dev/null)
```
Comment (lukebakken) [inline Makefile:6]: Could be built-in `dir` function instead of `shell dirname`.

**Comment by JeetKunDoug** [comment]
:+1: Thanks for putting this in - may help people not build broken things.

**Comment by JeetKunDoug** [comment]
@lukebakken can you please recreate this targeting the 2.0 branch for merge? Will get it forward-merged from there.

**Pull request closed without merging**